lib/mongoid/criteria/marshalable.rb

```diff
@@ -11,7 +11,7 @@ module Marshalable
       #
       # @return [ Array<Object> ] The dumped data.
       def marshal_dump
-        data = [ klass, driver, inclusions, documents, strategy, negating ]
+        data = [ klass, :mongo, inclusions, documents, strategy, negating ]
```

Comment:
Yes, but the `load` method shouldn't set `@driver` anymore. Could you also add a note that `:mongo` is written here for backwards compatibility with Mongoid 7 and earlier serializations please?

Reply:
done and done